### PR TITLE
Fix null ref in declaration name computation

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationName/DeclarationNameInfo.cs [hunk 1]
```diff
@@ -23,14 +23,17 @@ namespace Microsoft.CodeAnalysis.CSharp.Completion.Providers.DeclarationName
...
+        private readonly ImmutableArray<SymbolKindOrTypeKind> _possibleSymbolKinds;
+
         public readonly DeclarationModifiers Modifiers;
         public readonly Accessibility? DeclaredAccessibility;
 
         public readonly ITypeSymbol? Type;
         public readonly IAliasSymbol? Alias;
         public readonly ISymbol? Symbol;
 
+        public ImmutableArray<SymbolKindOrTypeKind> PossibleSymbolKinds => _possibleSymbolKinds.NullToEmpty();
```
Comment (ryzngard) [inline src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationName/DeclarationNameInfo.cs:35]: Is there any reason to do this as a getter instead of just doing this in the constructor and assigning once?

File: src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationName/DeclarationNameInfo.cs [hunk 2]
```diff
@@ -39,15 +42,15 @@ namespace Microsoft.CodeAnalysis.CSharp.Completion.Providers.DeclarationName
...
+            _possibleSymbolKinds = possibleSymbolKinds;
             DeclaredAccessibility = accessibility;
             Modifiers = declarationModifiers;
             Type = type;
             Alias = alias;
             Symbol = symbol;
         }
 
-        internal static async Task<NameDeclarationInfo> GetDeclarationInfoAsync(Document document, int position, CancellationToken cancellationToken)
+        public static async Task<NameDeclarationInfo> GetDeclarationInfoAsync(Document document, int position, CancellationToken cancellationToken)
```
Comment (ryzngard) [inline src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationName/DeclarationNameInfo.cs:53]: this seems unrelated. I'm guessing just cleanup because the type is internal?

**Review by ryzngard — APPROVED** [review]

File: src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationName/DeclarationNameInfo.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationName/DeclarationNameInfo.cs:35]: this is a struct, so the constructor is not always called.  this makes us follow the pattern used in things like SymbolInfo where the default-struct is the proper representation of 'no value'.

File: src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationName/DeclarationNameInfo.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationName/DeclarationNameInfo.cs:53]: yup :)
